### PR TITLE
feature: support custom lua TT filters.

### PR DESCRIPTION
**Pull request by ingydotnet**
Example:
```lua
lemplate = require('my.templates')
lemplate.filters['upper'] = function(str, args)
    return string.upper(str)
end
lemplate.process('[% "i like pie" | upper %]')
```

### CODE REVIEW

**Review by agentzh — APPROVED** [review]

**Comment by agentzh** [comment]
@doujiang24 Have a look at this :)

@ingydotnet Thanks!

**Comment by doujiang24** [comment]
@ingydotnet Thanks, works fine on my side. one question: seems we should also update [lemplate](https://github.com/openresty/lemplate/blob/master/lemplate).

**Comment by ingydotnet** [comment]
Fixed. Please Review.

File: lib/Lemplate.pm [hunk 1]
```diff
@@ -538,18 +538,35 @@ local function stash_set(stash, k, v, default)
...
-            if name == "html" then
-                s = gsub(s, "&", '&amp;', "jo")
-                s = gsub(s, "<", '&lt;', "jo");
-                s = gsub(s, ">", '&gt;', "jo");
-                s = gsub(s, '"', '&quot;', "jo"); -- " end quote for emacs
-                return s
+            local f = _M.filters[name]
+            if f then
+                return f(s, params)
+            else
```
Comment (agentzh) [inline lib/Lemplate.pm]: We should not use `else` when the previous branch never falls through :) I think we encountered this before ;)

**Review by agentzh — CHANGES_REQUESTED** [review]

File: lib/Lemplate.pm (hunk 1 above)
Comment (ingydotnet) [reply lib/Lemplate.pm]: Fixed. Please review.

**Comment by agentzh** [comment]
@ingydotnet Will you rebase to the latest master and resolve those conflicts? Then I think it's good to merge. Thanks a lot!

**Comment by ingydotnet** [comment]
Done. Please review.

**Comment by agentzh** [comment]
@ingydotnet Merged. Thanks!

**Pull request closed without merging**